src/models/transactions/common.ts

```diff
@@ -55,13 +55,18 @@ const SIGNER_SIZE = 3;
 
 function isSigner(obj: unknown): boolean {
   // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- Only used by JS
-  const signer = obj as Record<string, unknown>;
-  return (
-    Object.keys(signer).length === SIGNER_SIZE &&
-    typeof signer.Account === "string" &&
-    typeof signer.TxnSignature === "string" &&
-    typeof signer.SigningPubKey === "string"
-  );
+  const signerWrapper = obj as Record<string, unknown>;
+  if (signerWrapper.Signer !== undefined) {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- Only used by JS and Signer is previously unknown
+    const signer = signerWrapper.Signer as Record<string, unknown>;
+    return (
+      Object.keys(signer).length === SIGNER_SIZE &&
+      typeof signer.Account === "string" &&
+      typeof signer.TxnSignature === "string" &&
+      typeof signer.SigningPubKey === "string"
+    );
+  }
+  return false;
```

Comment:
i think this will be easier to read if you invert the conditional, since all your heavy lifting is within the if. like so:

```typescript
function isSigner(obj: unknown): boolean {
  const signerWrapper = obj as Record<string, unknown>
  if (signerWrapper.Signer === undefined) {
    return false
  }
  const signer = signerWrapper.Signer as Record<string, unknown>
  return (
    Object.keys(signer).length === SIGNER_SIZE &&
    typeof signer.Account === "string" &&
    typeof signer.TxnSignature === "string" &&
    typeof signer.SigningPubKey === "string"
  )
}
```